File: app/backend/java/auth-service/src/main/java/com/srbms/authservice/model/Driver.java
package com.srbms.authservice.model;

import jakarta.persistence.*;

@Entity
@Table(name = "driver")
public class Driver {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "driver_id")
    private Integer driverId;

    @OneToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Column(name = "license_no", nullable = false)
    private String licenseNo;

    @Column(name = "status")
    private String status = "unverified";

    @Column(name = "license_pdf_url", columnDefinition = "LONGTEXT")
    private String licensePdfUrl;

    public Driver() {}

    public Driver(User user, String licenseNo, String status) {
        this.user = user;
        this.licenseNo = licenseNo;
        this.status = status;
    }

    public Driver(User user, String licenseNo, String status, String licensePdfUrl) {
        this.user = user;
        this.licenseNo = licenseNo;
        this.status = status;
        this.licensePdfUrl = licensePdfUrl;
    }

    public Integer getDriverId() {
        return driverId;
    }

    public void setDriverId(Integer driverId) {
        this.driverId = driverId;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public String getLicenseNo() {
        return licenseNo;
    }

    public void setLicenseNo(String licenseNo) {
        this.licenseNo = licenseNo;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public String getLicensePdfUrl() {
        return licensePdfUrl;
    }

    public void setLicensePdfUrl(String licensePdfUrl) {
        this.licensePdfUrl = licensePdfUrl;
    }
}

File: app/backend/.net/SmartRide.AuthService/Models/Entities/Driver.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SmartRide.AuthService.Models.Entities;

[Table("driver")]
public class Driver
{
    [Key]
    [Column("driver_id")]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public int DriverId { get; set; }

    [Column("user_id")]
    public int UserId { get; set; }

    [ForeignKey(nameof(UserId))]
    public User? User { get; set; }

    [Required]
    [Column("license_no")]
    [StringLength(50)]
    public string LicenseNo { get; set; } = string.Empty;

    [Column("status")]
    [StringLength(50)]
    public string Status { get; set; } = "unverified";

    [Column("license_pdf_url", TypeName = "LONGTEXT")]
    public string? LicensePdfUrl { get; set; }

    public Driver() { }

    public Driver(User user, string licenseNo, string status)
    {
        User = user;
        UserId = user.UserId;
        LicenseNo = licenseNo;
        Status = status;
    }

    public Driver(User user, string licenseNo, string status, string? licensePdfUrl)
    {
        User = user;
        UserId = user.UserId;
        LicenseNo = licenseNo;
        Status = status;
        LicensePdfUrl = licensePdfUrl;
    }
}
